staging/src/k8s.io/client-go/tools/cache/reflector_test.go

```diff
@@ -759,12 +764,15 @@ func TestReflectorFullListIfTooLarge(t *testing.T) {
 	// may be synced to a different version and they will never converge.
 	// TODO: We should use etcd progress-notify feature to avoid this behavior but until this is
 	// done we simply try to relist from now to avoid continuous errors on relists.
-	stopCh = make(chan struct{})
-	if err := r.ListAndWatch(stopCh); err != nil {
-		t.Fatal(err)
+	for i := 1;  i<=2; i++ {
+		// relist twice to cover the two variants of TooLargeResourceVersion api errors
+		stopCh = make(chan struct{})
```

Comment:
create stopCh just once

Reply:
I think each iteration needs a new channel otherwise it would mean passing an already closed channel to  `ListAndWatch` on the second iteration. Reusing the same channel make the test fail like this:

```
W0831 15:24:13.372859   82189 reflector.go:424] testing/testing.go:1108: watch of *v1.Pod ended with: very short watch: testing/testing.go:1108: Unexpected watch close - watch lasted less than a second and no items received
W0831 15:24:13.375340   82189 reflector.go:424] testing/testing.go:1108: watch of *v1.Pod ended with: expected resync reached
--- FAIL: TestReflectorFullListIfTooLarge (0.00s)
```

Reply:
aah ok - we're closing the channel in line 725. Thanks for checking.